### PR TITLE
Factorio: fix website multitracker

### PR DESCRIPTION
**Pull request by Berserker66**
## What is this fixing or adding?
https://discord.com/channels/731205301247803413/1145166962910040104

## How was this tested?
minimally

### CODE REVIEW

File: WebHostLib/tracker.py [hunk 1]
```diff
@@ -1447,18 +1447,19 @@ def get_multiworld_tracker(tracker: UUID):
...
-def get_Factorio_multiworld_tracker(tracker: UUID):
-    data = _get_multiworld_tracker_data(tracker)
-    if not data:
-        abort(404)
-
-    data["inventory"] = _get_inventory_data(data)
-    data["enabled_multiworld_trackers"] = get_enabled_multiworld_trackers(data["room"], "Factorio")
+        data["inventory"] = _get_inventory_data(data)
+        data["enabled_multiworld_trackers"] = get_enabled_multiworld_trackers(data["room"], "Factorio")
+        data["item_name_to_id"] =  games["Factorio"]["location_name_to_id"]
```
Comment (remyjette) [inline WebHostLib/tracker.py:1460]: Should this be
```suggestion
        data["item_name_to_id"] =  games["Factorio"]["item_name_to_id"]
```

**Comment by remyjette** [comment]
I just tested it with progressive techs on and off, seems to be working correctly now. Thanks!

**Review by remyjette — APPROVED** [review]

**Review by black-sliver — COMMENTED** [review]
This should work for current seeds, however if we ever change IDs again, this will break. item_name_to_id really should be fetched from the actual games package the multidata uses.

**Comment by Berserker66** [comment]
yes

**Comment by black-sliver** [comment]
> yes

so, do you want to merge this as-is or fix it in one swoop?

**Comment by Berserker66** [comment]
I would prefer to fix that too, if I can find the time to do so.

File: WebHostLib/tracker.py [hunk 2]
```diff
@@ -1436,6 +1439,17 @@ def _get_inventory_data(data: typing.Dict[str, typing.Any]) -> typing.Dict[int,
     return inventory
 
 
+def _get_named_inventory(inventory: typing.Dict[int, int], custom_items: typing.Dict[int, str] = None) \
+        -> typing.Dict[str, int]:
+    """slow"""
```
Comment (black-sliver) [inline WebHostLib/tracker.py:1444]: Could probably be lazy?

**Review by black-sliver — APPROVED** [review]
Looks fine to me, however I have **not** tested it. Maybe someone else can?

**Comment by Berserker66** [comment]
nope, forgotten to do

**Comment by remyjette** [comment]
> Looks fine to me, however I have **not** tested it. Maybe someone else can?

I just tested it and things appear to be working correctly.

**Pull request merged**